File: gs/src/gate/Module.java
package gate;

import common.IHandler;
import common.IModule;
import gs.Config;
import io.netty.channel.nio.NioEventLoopGroup;
import msg._Refs_;

/**
 * Created by HuangQiang on 2017/6/3.
 */
public enum Module implements IModule {
    Ins;


    @Override
    public void start() {
        Config conf = Config.getIns();
        NioEventLoopGroup boss = new NioEventLoopGroup();
        {
            GateClient.Conf gc = conf.getGateConf();
            gc.stubs = _Refs_.gate_server;
            gc.bossGroup = boss;
            GateClient.start(gc);
        }
    }

    @Override
    public IHandler getHandler() {
        return Handler.Ins;
    }
}

File: gs/src/game/mall/Module.kt
package game.mall

import common.IHandler
import common.IModule
import game.event.EvtMgr

/**
 * Created by HuangQiang on 2017/6/5.
 */
object Module : IModule {
    override fun start() {
        EvtMgr.add(game.event.Login::class.java, this, this::onEvent)
        EvtMgr.add(game.event.Logout::class.java, this, this::onEvent)
    }

    override fun getHandler(): IHandler {
        return Handler
    }

    private fun onEvent(evt : game.event.Login) {
        println("==== ${evt.roleid} login ==")
    }

    private fun onEvent(evt : game.event.Logout) {
        println("==== ${evt.roleid} login ==")
    }
}